src/Simplex/Chat/Store.hs

```diff
@@ -3846,24 +3847,33 @@ updateDirectChatItem db userId contactId itemId newContent live msgId_ = do
 
 updateDirectChatItem_ :: forall d. (MsgDirectionI d) => DB.Connection -> UserId -> Int64 -> ChatItemId -> CIContent d -> Bool -> UTCTime -> ExceptT StoreError IO (ChatItem 'CTDirect d)
 updateDirectChatItem_ db userId contactId itemId newContent live currentTs = do
-  ci@ChatItem {meta = CIMeta {itemEdited, itemLive}} <- liftEither . correctDir =<< getDirectChatItem db userId contactId itemId
+  ci@ChatItem {meta = meta@CIMeta {itemEdited, itemTimed, itemLive}} <- liftEither . correctDir =<< getDirectChatItem db userId contactId itemId
   let newText = ciContentToText newContent
       edited' = itemEdited || (itemLive /= Just True)
       live' = (live &&) <$> itemLive
+      delAt' = ciLiveDeleteAt meta live currentTs
+      timed' = (\timed -> timed {deleteAt = delAt'}) <$> itemTimed
   liftIO $ do
     DB.execute
       db
       [sql|
         UPDATE chat_items
-        SET item_content = ?, item_text = ?, item_deleted = 0, item_edited = ?, item_live = ?, updated_at = ?
+        SET item_content = ?, item_text = ?, item_deleted = 0, item_edited = ?, timed_delete_at = ?, item_live = ?, updated_at = ?
         WHERE user_id = ? AND contact_id = ? AND chat_item_id = ?
       |]
-      (newContent, newText, edited', live', currentTs, userId, contactId, itemId)
-  pure ci {content = newContent, meta = (meta ci) {itemText = newText, itemEdited = edited', itemLive = live'}, formattedText = parseMaybeMarkdownList newText}
+      (newContent, newText, edited', delAt', live', currentTs, userId, contactId, itemId)
+  pure ci {content = newContent, meta = meta {itemText = newText, itemEdited = edited', itemTimed = timed', itemLive = live'}, formattedText = parseMaybeMarkdownList newText}
   where
     correctDir :: CChatItem c -> Either StoreError (ChatItem c d)
     correctDir (CChatItem _ ci) = first SEInternalError $ checkDirection ci
 
+-- the condition to enable the timed deletion when the item that was live is updated
+ciLiveDeleteAt :: CIMeta d -> Bool -> UTCTime -> Maybe UTCTime
```

Comment:
maybe move this logic to Chat, and add an explicit check item went from live to not in startUpdatedTimedItemThread as a sanity check

Reply:
I guess getDirectChatItem would have to be moved from Store update functions to Chat as well then

Reply:
That’s the reason I didn’t want to change it all - it’s ok as is

Reply:
In any case if it’s to be done it should be a separate PR

Reply:
startUpdatedTimedItemThread - it has the check that deleteAt was added, why does it matter for what reason?

Reply:
btw items are already being read in Chat before update both on send and receive